### PR TITLE
Add support for stream query cursor

### PR DESCRIPTION
**Pull request by graetzer**
#99 should be merged first, then travis will also run green again

### CODE REVIEW

**Comment by dgallion1** [comment]
This looks like the feature I need.
I'm relaying records to browsers, direct javascript results are much faster.
Having to marshal the data in/out of goLang can't be good.
Thanks.

File: cursor.go [hunk 1]
```diff
@@ -44,6 +65,11 @@ type Cursor interface {
 
 	// Count returns the total number of result documents available.
 	// A valid return value is only available when the cursor has been created with a context that was
-	// prepare with `WithQueryCount`.
+	// prepared with `WithQueryCount` and not with `WithStream`.
 	Count() int64
+
+	// Return execution statistics for this cursor. This might not
```
Comment (ewoutp) [inline cursor.go]: Make sure to start comment with `Statistics`

File: cursor_impl.go [hunk 2]
```diff
@@ -52,11 +52,32 @@ type cursor struct {
 	closeMutex  sync.Mutex
 }
 
+type cursorStats struct {
```
Comment (ewoutp) [inline cursor_impl.go]: Please document this type

File: cursor_impl.go [hunk 3]
```diff
@@ -52,11 +52,32 @@ type cursor struct {
...
+}
+
 type cursorData struct {
 	Count   int64        `json:"count,omitempty"`   // the total number of result documents available (only available if the query was executed with the count attribute set)
 	ID      string       `json:"id"`                // id of temporary cursor created on the server (optional, see above)
 	Result  []*RawObject `json:"result,omitempty"`  // an array of result documents (might be empty if query has no results)
 	HasMore bool         `json:"hasMore,omitempty"` // A boolean indicator whether there are more results available for the cursor on the server
+	Extra   struct {
+		Stats cursorStats `json:"stats,omitempty"`
+		// TODO profile, warnings etc
```
Comment (ewoutp) [inline cursor_impl.go]: Is this TODO for this PR? 
If not make an issue for it

File: cursor_impl.go [hunk 4]
```diff
@@ -52,11 +52,32 @@ type cursor struct {
...
+	} `json:"extra,omitempty"`
```
Comment (ewoutp) [inline cursor_impl.go]: The `omitempty` will be ignored here because the struct is a value type, not a reference to a struct.

File: cursor.go [hunk 5]
```diff
@@ -27,6 +27,27 @@ import (
...
+	ScannedIndex() int64
+	// the total number of documents that were removed after executing a filter condition in a FilterNode
+	Filtered() int64
+	// Returns the numer of results before the last LIMIT in the query was applied.
+	// A valid return value is only available when the has been created with a context that was
+	// prepared with `WithFullCount`. Additionally this will also not return a valid value if
+	// the context was prepared with `WithStream`.
+	FullCount() int64
+	// query execution time (wall-clock time). value will be set from the outside
+	ExecutionTime() float64
```
Comment (ewoutp) [inline cursor.go]: Should we change the return type to `time.Duration`? IMO we should
If not, at least specify the unit of the value.

**Pull request merged**